sw/device/tests/dif_gpio/meson.build

```diff
@@ -0,0 +1,25 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+gtest_inc = include_directories(
+  '/tools/googletest/gtest',
+  '/tools/googletest/gmock'
+)
+
+dif_gpio_test = executable(
+  'dif_gpio_test',
+  sources: [
+    hw_ip_gpio_reg_h,
+    'mock_mmio.cc',
+    'dif_gpio_test.cc',
+    '../../lib/dif/dif_gpio.c',
+    '../../lib/base/mmio.c'],
+  include_directories: gtest_inc,
+  link_args: ['-L/tools/googletest/lib', '-lgtest', '-lgtest_main', '-lgmock', '-lpthread'],
+  native: true,
+  cpp_args: ['-DMOCK_MMIO'],
+  c_args: ['-DMOCK_MMIO']
+)
```

Comment:
You should factor out gtest and mock_mmio.cc into separate dependencies, like in #1298.